### ISSUE TITLE
check why kubectl exec does not work using docker 18.06

### ISSUE THREAD

**Issue by chavafg** [kata-containers/tests#520]
```
1..1
not ok 1 Check CPU constraints
# (in test file k8s-cpu-ns.bats, line 34)
#   `total_cpus_container=$(sudo -E kubectl exec $pod_name -c $container_name nproc)' failed
# pod "constraints-cpu-test" created
# Flag --show-all has been deprecated, will be removed in an upcoming release
# Flag --show-all has been deprecated, will be removed in an upcoming release
# Flag --show-all has been deprecated, will be removed in an upcoming release
# Flag --show-all has been deprecated, will be removed in an upcoming release
# constraints-cpu-test   1/1       Running   0          16s
# error: unable to upgrade connection: 404 page not found
# Error from server (NotFound): deployments.extensions "constraints-cpu-test" not found
# NAME                   READY     STATUS    RESTARTS   AGE
# constraints-cpu-test   1/1       Running   0          32s
Makefile:44: recipe for target 'kubernetes' failed
```

**Comment by bergwolf** [comment]
@chavafg AFAICT, the runtime repo specifies `v18.03-ce` docker version. Could you check why `docker-ce 18.06` is installed instead?

**Comment by chavafg** [comment]
@bergwolf yes, the runtime repo specifies `18.03`, but the VM initialization cloud-init file, also installs `docker-ce`:
https://github.com/kata-containers/ci/blob/master/jenkins/org.jenkinsci.plugins.configfiles.GlobalConfigFiles.xml#L17-L22

There, we do not specify a version, since we do not have the runtime repo cloned in that state, so it is installing the latest one. 
Instead of hardcoding the version there, I opened #524  that will install the correct version if it is not already installed.

We are installing docker-ce in the VM startup and adding the current non-root user to the docker group since some unit tests run docker using the non-root user. The reason for doing it in the VM startup is that if we do this on the same bash session of the setup and test execution, we won't be able to use docker with the non-root user.

**Comment by bergwolf** [comment]
@chavafg Thanks for the details. It seems to me that we should stick to a fixed version of softwares (at least docker) when we install them in https://github.com/kata-containers/ci/blob/master/jenkins/org.jenkinsci.plugins.configfiles.GlobalConfigFiles.xml. WDYT?

**Comment by sboeuf** [comment]
@bergwolf yes that's the plan to always stick to a specific version so that we can easily identify where the errors are coming from. We got an issue with this one because we didn't realize the version was overridden by the VM, by installing the latest one.

In the meantime, we should fix the port issue introduced with docker 18.06, because we probably want to move to the latest Docker rather sooner than later.

**Comment by sboeuf** [comment]
Issue being discussed here: https://github.com/kata-containers/tests/issues/521#issuecomment-407216227
